dialog/file.go

```diff
@@ -499,10 +512,12 @@ func (f *fileDialog) setView(view viewLayout) {
 		grid := widget.NewGridWrap(count, template, update)
 		grid.OnSelected = choose
 		f.files = grid
+		fyne.CurrentApp().Preferences().SetString("fileDialogViewLayout", "grid")
 	} else {
 		list := widget.NewList(count, template, update)
 		list.OnSelected = choose
 		f.files = list
+		fyne.CurrentApp().Preferences().SetString("fileDialogViewLayout", "list")
 	}
```

Comment:
With my suggestion to not store strings in the preferences, you can hoist it out of the if-statement.
```suggestion
	} else {
		list := widget.NewList(count, template, update)
		list.OnSelected = choose
		f.files = list
	}
	
	fyne.CurrentApp().Preferences().SetInt("fileDialogViewLayout", int(view))
```